test/NuGet.Clients.Tests/NuGet.VisualStudio.Implementation.Test/Extensibility/VsGlobalPackagesInitScriptExecutorTests.cs

```diff
@@ -1,18 +1,27 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
 using System;
+using System.Collections.Generic;
 using System.Threading.Tasks;
 using Moq;
 using NuGet.PackageManagement.VisualStudio;
+using NuGet.Packaging.Core;
+using NuGet.VisualStudio.Telemetry;
 using Xunit;
 
 namespace NuGet.VisualStudio.Implementation.Test
 {
     public class VsGlobalPackagesInitScriptExecutorTests
     {
+        // known/expected errors should not be reported to telemetry, hence use MockBehavior.Strict
+        private Mock<INuGetTelemetryProvider> _telemetryProvider = new Mock<INuGetTelemetryProvider>(MockBehavior.Strict);
+
         [Fact]
         public async Task NullPackageId()
         {
             var scriptExecutor = new Mock<IScriptExecutor>();
-            var executor = new VsGlobalPackagesInitScriptExecutor(scriptExecutor.Object);
+            var executor = new VsGlobalPackagesInitScriptExecutor(scriptExecutor.Object, _telemetryProvider.Object);
```

Comment:
Should you add an Assertion to each of these tests that proves that the exception was recorded in telemetry (or not recorded, if that `Strict` means it explicitly should not)

Reply:
`Strict` mode means that if any method that was not setup on the mock object is called, then an exception is thrown (NotImplementedException? I can't remember which exception exactly). So, it's like validating that no methods were called, but with less effort. Although any catch-all block will catch those exceptions.

The only test in this class that does actually post a fault does validate that the telemetry method was called ([line 83](https://github.com/NuGet/NuGet.Client/pull/3775/files/59157cc7f88f9f26108a78ab6352493df926e5fa#diff-de831b35e6318dba442d5645b3089541722c1491cee618a606d8b88585a00673R83))